hdf5/src/hl/references/mod.rs

```diff
@@ -0,0 +1,53 @@
+use crate::internal_prelude::*;
```

Comment:
The reason this was done as a folder was to separate the two different reference types into their own submodules to aid readability and different imports based on the version. We can do this but then they will have to be submodules in the same file? Or am I missing an alternative?

Reply:
I agree that the directory structure makes sense. A file named `foo/mod.rs` can be replaced by a `foo.rs`, making the toplevel a bit clearer instead of a "magic" mod.rs file.

The only change requested is `mv references/mod.rs references.rs`, the rest works as is

Reply:
Ah I did not realize that worked! Perfect I'll make that change

Reply:
Done in [eb70091](https://github.com/metno/hdf5-rust/pull/36/commits/eb70091100bb9707a78f2f2fff737e5050e5cb1e) below